src/porepy/applications/boundary_conditions/model_boundary_conditions.py

```diff
@@ -290,3 +291,366 @@ def bc_values_displacement(self, bg: pp.BoundaryGrid) -> np.ndarray:
+
+        """
+        domain_sides = self.domain_boundary_sides(sd)
+        box = self.domain.bounding_box
+
+        # Point 1 is on the center top of the west boundary, having min x coordinate and
+        # mean y coordinate.
+        x_mean = 0.5 * (box["xmax"] + box["xmin"])
+        y_mean = 0.5 * (box["ymax"] + box["ymin"])
+        z_max = box["zmax"]
+        point_1 = np.array([box["xmin"], y_mean, z_max])
+        pts = sd.face_centers[:, domain_sides.west]
+        ind_1 = domain_sides.west.nonzero()[0][
+            np.argmin(pp.distances.point_pointset(point_1, pts))
+        ]
+        # Point 2 is on the center top of the east boundary.
+        point_2 = np.array([box["xmax"], y_mean, z_max])
+        pts = sd.face_centers[:, domain_sides.east]
+        ind_2 = domain_sides.east.nonzero()[0][
+            np.argmin(pp.distances.point_pointset(point_2, pts))
+        ]
+        # Point 3 is on the center top of the south boundary, having min y coordinate
+        # and mean x coordinate.
+        point_3 = np.array([x_mean, box["ymin"], z_max])
+        pts = sd.face_centers[:, domain_sides.south]
+        ind_3 = domain_sides.south.nonzero()[0][
+            np.argmin(pp.distances.point_pointset(point_3, pts))
+        ]
+        return [ind_1, ind_2, ind_3]
+
+
+class GravityMagnitude:
+    """Mixin class for gravity magnitude computation for different materials.
+
+    See also :class:`~porepy.models.constitutive_laws.GravityForce`, which returns the
+    gravity force as an AD operator.
+
+    """
+
+    fluid: pp.Fluid
+    """Fluid model associated with the flow problem."""
+    solid: pp.SolidConstants
+    """Solid model associated with the mechanics problem."""
+    units: pp.Units
+    """Model units."""
+
+    def gravity_force_magnitude(
+        self, material: Literal["fluid", "solid", "bulk"]
+    ) -> float:
+        """Compute gravity force magnitude for a given region.
+
+        Parameters:
+            material: Material for which to compute gravity force. Can be "bulk",
+                "fluid" or "solid".
+
+        Returns:
+            Gravity force magnitude for the specified material.
+        """
+        g = self.units.convert_units(pp.GRAVITY_ACCELERATION, "m * s^-2")
+        if material == "bulk":
+            phi = self.solid.porosity
+            gravity = g * (
+                self.fluid.reference_component.density * phi
+                + (1 - phi) * self.solid.density
+            )
+        elif material == "fluid":
+            gravity = g * self.fluid.reference_component.density
+        elif material == "solid":
+            gravity = g * self.solid.density
+        else:
+            raise ValueError(f"Unknown region '{material}' for gravity force.")
+        return gravity
+
+
+class LithostaticBoundaryStressValues(GravityMagnitude):
+    """Boundary conditions for the mechanics with lithostatic stress on all
+    boundaries.
```

Comment:
Mention explicitly that the principal stress axes are assumed to coincide with the coordinate axes?